Detectors/TPC/reconstruction/src/TPCCATracking.cxx

```diff
@@ -0,0 +1,156 @@
+  deinitialize();
+}
+
+int TPCCATracking::initialize(const char* options) {
+#ifndef HAVE_O2_TPCCA_TRACKING_LIB
+  LOG(FATAL) << "O2 compiled without TPC CA Tracking library, cannot run TPC CA Tracker\n";
+  return (1);
+#else
+  mTrackingCAO2Interface.reset(new AliHLTTPCCAO2Interface);
+  int retVal = mTrackingCAO2Interface->Initialize(options);
+  if (retVal) {
+    mTrackingCAO2Interface.reset();
+  } else {
+    mClusterData_UPTR.reset(new AliHLTTPCCAClusterData[36]);
+    mClusterData = mClusterData_UPTR.get();
+  }
+  return (retVal);
+#endif
+}
+
+void TPCCATracking::deinitialize() {
+  mTrackingCAO2Interface.reset();
+  mClusterData_UPTR.reset();
+  mClusterData = NULL;
+}
+
+int TPCCATracking::runTracking(const TClonesArray* inputClusters, std::vector<TrackTPC>* outputTracks) {
+  if (mTrackingCAO2Interface == nullptr) return (1);
+  int retVal = 0;
+
+#ifdef HAVE_O2_TPCCA_TRACKING_LIB
+  const static ParameterDetector &detParam = ParameterDetector::defaultInstance();
+  const static ParameterGas &gasParam = ParameterGas::defaultInstance();
+  const static ParameterElectronics &elParam = ParameterElectronics::defaultInstance();
+  
+  const AliHLTTPCGMMergedTrack* tracks;
+  int nTracks;
+  const unsigned int* trackClusterIDs;
+
+  int nClusters[36] = {};
+  for (Int_t icluster = 0; icluster < inputClusters->GetEntries(); ++icluster) {
+    Cluster& cluster = *static_cast<Cluster*>(inputClusters->At(icluster));
+    const Sector sector = CRU(cluster.getCRU()).sector();
+    nClusters[sector.getSector()]++;
+  }
+  for (int i = 0; i < 36; i++) {
+    mClusterData[i].StartReading(i, nClusters[i]);
+  }
+  int nClustersConverted = 0;
+  for (Int_t icluster = 0; icluster < inputClusters->GetEntries(); ++icluster) {
+    Cluster& cluster = *static_cast<Cluster*>(inputClusters->At(icluster));
+    const CRU cru(cluster.getCRU());
+    const Sector sector = cru.sector();
+    AliHLTTPCCAClusterData& cd = mClusterData[sector.getSector()];
+    AliHLTTPCCAClusterData::Data& hltCluster = cd.Clusters()[cd.NumberOfClusters()];
+
+    // ===| mapper |==============================================================
+    Mapper& mapper = Mapper::instance();
+    const PadRegionInfo& region = mapper.getPadRegionInfo(cru.region());
+    const int rowInSector = cluster.getRow() + region.getGlobalRowOffset();
+    const float padY = cluster.getPadMean();
+    const int padNumber = int(padY);
+    const GlobalPadNumber pad = mapper.globalPadNumber(PadPos(rowInSector, padNumber));
+    const PadCentre& padCentre = mapper.padCentre(pad);
+    const float localY = padCentre.Y() - (padY - padNumber - 0.5) * region.getPadWidth();
+    const float localYfactor = (cru.side() == Side::A) ? -1.f : 1.f;
+    float zPositionAbs = cluster.getTimeMean()*elParam.getZBinWidth()*gasParam.getVdrift();
+    if (!mTrackingCAO2Interface->GetParamContinuous()) zPositionAbs = detParam.getTPClength() - zPositionAbs;
+
+    Point2D<float> clusterPos(padCentre.X(), localY);
```

Comment:
The code related to the transformations is not specific to the `TPCCATracking` class, this can probably be moved to some TPC base functionality (or it's existing there already).

Reply:
Hi,
indeed this could be done. But what is in there is anyway only a mockup, such that people can use the tracking for tests right now. In the end, we need a proper transformation, and even further, for processing timeframes, the tracker will anyway get time coordinate instead of Z.

Reply:
I Matthias and David, indeed there transformations should go elsewhere at some point. I'll look into this soon and also discuss with David and perhaps Ruben.